UPGRADE-PHP-TO-TWIG-TEMPLATES.md

```diff
@@ -0,0 +1,83 @@
+?>
+
+// ROUTING
+<a href="<?php echo $view['router']->path('/emails', ['objectAction' => 'batchDelete']); ?">Hello world!</a>
+
+// FORMS
+<?php echo $view['form']->start($form); ?>
+<?php echo $view['form']->row($form['email']); ?>
+<?php echo $view['form']->end($form); ?>
+
+// PAGE ACTIONS
+$view['slots']->set('actions', $view->render('MauticCoreBundle:Helper:page_actions.html.php', [
+    'item'            => $item,
+    'templateButtons' => [
+        'edit'   => $view['security']->hasEntityAccess($permissions['webhook:webhooks:editown'], $permissions['webhook:webhooks:editother'], $item->getCreatedBy()),
+        'clone'  => $permissions['webhook:webhooks:create'],
+        'delete' => $view['security']->hasEntityAccess($permissions['webhook:webhooks:deleteown'], $permissions['webhook:webhooks:deleteown'], $item->getCreatedBy()),
+    ],
+    'routeBase' => 'webhook',
+]));
+
+// TODO add more examples
+```
+
+Becomes
+
+```Twig
+{% extends 'MauticCoreBundle:Default:content.html.twig' %}
+
+{% block headerTitle %}{% trans %}mautic.webhook.webhooks{% endtrans %}{% endblock %}
+{% block mauticContent %}mauticWebhook{% endblock %}
+
+{# ROUTING #}
+<a href="{{ path('/emails', {objectAction: 'batchDelete'}) }}">Hello world!</a>
+
+{# FORMS #}
+{{ form_start(form) }}
+{{ form_row(form.email) }}
+{{ form_end(form) }}
+
+{# PAGE ACTIONS #}
+{% block actions %}
+    {{- include(
+        'MauticCoreBundle:Helper:page_actions.html.twig', {
+            item: item,
+            templateButtons: {
+                'edit': securityHasEntityAccess(
+                    permissions['webhook:webhooks:editown'],
+                    permissions['webhook:webhooks:editother'],
+                    item.getCreatedBy()
+                ),
+                'clone': permissions['webhook:webhooks:create'],
+                'delete': securityHasEntityAccess(
+                    permissions['webhook:webhooks:deleteown'],
+                    permissions['webhook:webhooks:deleteother'],
+                    item.getCreatedBy()
+                )
+            },
+            routeBase: 'webhook'
+    }) -}}
+{% endblock %}
+
+{# TODO add more examples #}
+```
+
+## Random notes
+
+- `strict_variables` is enabled in dev mode (`index_dev.php`) to help you prevent bugs in your code. TODO should we enable this in prod too???
+- If you extend `MauticCoreBundle:Default:content.html.twig`, everything HAS to be in blocks. Trying to put any HTML elements outside a block will fail with the following error:
+
+    > A template that extends another one cannot include content outside Twig blocks.
+- You're probably used to writing `if !empty($variable) {}` in PHP. That checks if the variable is set and whether it is not empty. In Twig, you explicity have to write `if variable is defined and variable is not empty`. It's a lot more descriptive, and we might want to create a custom Twig function like `notEmpty(variable)`.
```

Comment:
This one comes back many times. Writing `if variable is defined and variable is not empty` is a lot. As mentioned [here](https://stackoverflow.com/a/48302655/4337765), we basically have the following options:
- By using the `default` filter: e.g. `variable|default('testvalue')`
  - This would simplify `set pull = pull is defined and pull is not empty ? pull : 'left'` to `set pull = pull|default('left')`
  - This would **not** help us to simplify `(attr.readonly is defined and attr.readonly is not empty) ? ' disabled' : ''`
- By setting `strict_variables` to false - I'd like to keep this on as it helps us more easily discover potential bugs in the code. It's currently only enabled in dev mode `index_dev.php` as mentioned in the separate note above, but I'd like to enable it for prod as well. Does the team have any thoughts on this? [Docs here](https://symfony.com/doc/current/reference/configuration/twig.html#strict-variables)
- By extending Twig. We can e.g. create a custom function `notEmpty(variable)` which would reduce `(attr.readonly is defined and attr.readonly is not empty)` to `notEmpty(attr.readonly)`.

My preference would be to use a combination of the first option (if we can set a default value) and the third. Curious to hear the team's thoughts on this!

Reply:
I'm 100% with you, Dennis. 1+2 seems to be the way to go. Strict variables is good to have. The stricter the better.

Reply:
I think that this convo is resolved, so I will mark it as so - this will allow us to merge the PR.

Reply:
Discussion with Dennis - there is some work to do here so pending that before we merge this PR.

Reply:
Just applied the `default` filter where I could (love it!) and enabled `strict_variables` for prod too. Resolving this conversation now 😊 

https://github.com/mautic/mautic/pull/11382/commits/b539a5a8284f4004ea65af51633c320d8f2447bc